rueidisotel/trace.go

```diff
@@ -27,12 +27,25 @@ var (
 
 var _ rueidis.Client = (*otelclient)(nil)
 
-// WithClient creates a new rueidis.Client with OpenTelemetry tracing enabled
+// WithClient creates a new rueidis.Client with OpenTelemetry tracing enabled.
 func WithClient(client rueidis.Client, opts ...Option) rueidis.Client {
-	o := &otelclient{client: client}
-	for _, fn := range opts {
-		fn(o)
+	o := &otelclient{
+		client:         client,
+		meterProvider:  otel.GetMeterProvider(),  // Defaults to global MeterProvider
+		tracerProvider: otel.GetTracerProvider(), // Defaults to global TracerProvider
 	}
+	for _, opt := range opts {
+		opt(o)
+	}
+
+	// Set the MeterProvider and TracerProvider if specified
+	if o.meterProvider != nil {
+		otel.SetMeterProvider(o.meterProvider)
+	}
+	if o.tracerProvider != nil {
+		otel.SetTracerProvider(o.tracerProvider)
+	}
```

Comment:
These will set the global providers. We should not do that.
We should instead do this after applying `opts`:
```go
	if o.meterProvider == nil {
		o.meterProvider = otel.GetMeterProvider()
	}
	if o.tracerProvider == nil {
		o. tracerProvider  = otel.GetTracerProvider()
	}
```

Reply:
sorry for late you mean https://github.com/redis/rueidis/blob/1defd2dba8930c58d3f790ca89381ba370fd4111/rueidisotel/trace.go#L55 ?